### PR TITLE
To bring parity with travis build add missing tasks to kokoro build.

### PR DESCRIPTION
**Pull request by rghetia**
- Each task is treated as separate job to run in parallel.
- single script linux_presumbit.sh performs all task based TASK env.

### CODE REVIEW

**Comment by sebright** [comment]
Let us know when this is ready for review.

File: buildscripts/kokoro/linux_presubmit.sh [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+case "$TASK" in
+  "CHECK_GIT_HISTORY")
+    python ./scripts/check-git-history.py
+    ;;
+  "BUILD")
+    ./gradlew clean assemble --stacktrace
+    ./gradlew check :opencensus-all:jacocoTestReport
+    ./gradlew verGJF
+    curl -s https://codecov.io/bash | bash
```
Comment (sebright) [inline buildscripts/kokoro/linux_presubmit.sh]: It looks like this command didn't succeed in https://source.cloud.google.com/results/invocations/6c4b2d41-79b7-4c1f-b455-8fe121688b97/log:

```
+ curl -s https://codecov.io/bash
+ bash
  _____          _
 / ____|        | |
| |     ___   __| | ___  ___ _____   __
| |    / _ \ / _` |/ _ \/ __/ _ \ \ / /
| |___| (_) | (_| |  __/ (_| (_) \ V /
 \_____\___/ \__,_|\___|\___\___/ \_/
                              Bash-0b37652
x> No CI provider detected.
    Testing inside Docker? http://docs.codecov.io/docs/testing-with-docker
    Testing with Tox? https://docs.codecov.io/docs/python#section-testing-with-tox
    project root: .
    Yaml not found, that's ok! Learn more at http://docs.codecov.io/docs/codecov-yaml
==> Running gcov in . (disable via -X gcov)
==> Python coveragepy not found
==> Searching for coverage reports in:
    + .
    -> Found 1 reports
==> Detecting git/mercurial file structure
==> Reading reports
    + ./all/build/reports/jacoco/test/jacocoTestReport.xml bytes=1162483
==> Appending adjustments
    http://docs.codecov.io/docs/fixing-reports
    -> No adjustments found
==> Gzipping contents
==> Uploading reports
    url: https://codecov.io
    query: branch=pull_branch&commit=fa9860512b982d7d3ca02f940c050f09cca4a12e&build=&build_url=&name=&tag=&slug=census-instrumentation%2Fopencensus-java&service=&flags=&pr=&job=
    -> Pinging Codecov
HTTP 400
Please provide the repository token to upload reports via `-t :repository-token`
```

File: scripts/check-git-history.py [hunk 2]
```diff
@@ -4,17 +4,26 @@
...
     # pull request.
     exit_if_pull_request_has_merge_commits()
     print 'Checked pull request history: SUCCEEDED'
   else:
     print 'Skipped history check.'
 
+def is_kokoro_presubmit_request():
+  '''Returns true if TASK is set to 'CHECK_GIT_HISTORY' indicate a pull request.'''
+  if 'TASK' in os.environ:
+    return os.environ['TASK'] == 'CHECK_GIT_HISTORY'
```
[the remark this answers is not in the document]
Comment (rghetia) [reply scripts/check-git-history.py]: yes, TASK is also set in Travis so "if 'TASK' .. " is not required.

File: buildscripts/kokoro/linux_presubmit.sh (hunk 1 above)
Comment (rghetia) [reply buildscripts/kokoro/linux_presubmit.sh]: I'll remove it for now and add it back when other commits are ready.

File: scripts/check-git-history.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (rghetia) [reply scripts/check-git-history.py]: I misunderstood your comment. I have modified the check for env var KOKORO_GITHUB_PULL_REQUEST_NUMBER.

**Comment by rghetia** [comment]
Regarding your @sebright general comments.
1. Build names are in kokoro config. I'll change that to reflect appropriate names.
2. There are two linux builds right now. I'll remove the old one.
3. As we discussed following is still missing.
- badge status
- build status on commit (build everything on commit)
- android build instead of jdk7.

File: buildscripts/kokoro/linux_presubmit.sh [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    # Get token from file located at
+    # $KOKORO_KEYSTORE_DIR/73495_codecov-auth-token
+    if [ -f $KOKORO_KEYSTORE_DIR/73495_codecov-auth-token ] ; then
+      export CODECOV_TOKEN=`cat $KOKORO_KEYSTORE_DIR/73495_codecov-auth-token`
+    fi
+    if [[ ! -v CODECOV_TOKEN ]]; then
+      echo "Environment variable CODECOV_TOKEN not set"
+      exit 1
+    fi
+    curl -s https://codecov.io/bash | bash
```
Comment (sebright) [inline buildscripts/kokoro/linux_presubmit.sh]: Should we fail the build if the script fails?  It looks like the script has a -Z option to exit with 1 when it fails.

Comment (rghetia) [reply buildscripts/kokoro/linux_presubmit.sh]: yes it should exit with 1 when it fails. I'll change it in my next commit.

**Pull request merged**